packages/glossary-plugin/src/components/glossary-sidebar.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from "react";
+import Definition from "./definition";
+import { IWordDefinition, ILearnerDefinitions } from "./types";
+
+import * as css from "./glossary-sidebar.scss";
+
+interface IGlossarySidebarProps {
+  definitions: IWordDefinition[];
+  learnerDefinitions: ILearnerDefinitions;
+}
+
+enum Filter {
+  AllWords,
+  WithUserDefinitionOnly
+}
+
+interface IGlossarySidebarState {
+  filter: Filter;
+}
+
+const nonEmptyHash = (hash: any) => {
+  return Object.keys(hash).length > 0;
+};
+
+export default class GlossarySidebar extends React.Component<IGlossarySidebarProps, IGlossarySidebarState> {
```

Comment:
FYI, I have taken to reducing redundancy in component definitions by using `IProps` and `IState` -- they're not exported so they only have to be unique within the file. We've been using this convention on CLUE.